spring-xd-messagebus-rabbit/src/main/java/org/springframework/xd/dirt/integration/rabbit/RabbitMessageBus.java

```diff
@@ -673,6 +682,29 @@ public void destroy() {
 		stopBindings();
 	}
 
+	@Override
+	public void doManualAck(LinkedList<MessageHeaders> messageHeadersList) {
+		Iterator<MessageHeaders> iterator = messageHeadersList.iterator();
+		Map<Object, Long> channelsToAck = new HashMap<Object, Long>();
+		while(iterator.hasNext()) {
+			MessageHeaders messageHeaders = iterator.next();
+			if (messageHeaders.containsKey(AmqpHeaders.CHANNEL)) {
+				Channel channel = (com.rabbitmq.client.Channel) messageHeaders.get(AmqpHeaders.CHANNEL);
+				Long deliveryTag = (Long) messageHeaders.get(AmqpHeaders.DELIVERY_TAG);
+				channelsToAck.put(channel, deliveryTag);
+			}
+		}
+		for (Map.Entry<Object, Long> entry : channelsToAck.entrySet()) {
+			try {
+				Channel channel = (Channel) entry.getKey();
+				channel.basicAck(entry.getValue(), true);
```

Comment:
You don't really need the `channel` variable here; you can concat these 2 lines.